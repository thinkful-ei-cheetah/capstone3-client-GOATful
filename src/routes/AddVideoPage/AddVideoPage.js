import React, { Component } from 'react'
import AddVideoForm from '../../components/AddVideoForm/AddVideoForm';
import VideoStorage from '../../services/video-storage';
import {errorCheckNewVideo, tagStringToArray, checkTime} from '../../Utils/Utils'

export default class AddVid extends Component {
  state = {
    title: "",
    tags: '',
    video_length: "",
    youtube_display_name: "",
    addError: null,
  }

  cssRouteClasses = {
    form: 'add-video-form-route',
    instructions: 'add-video-form-instructions',
    flex: 'flex-form-route',
    directionsBox: 'directions-route',
    inputBox: 'inputs-route',
    labelBox: 'label-route'
  }
  
  errorHandler = err => {
    this.setState({addError: err.error})
    console.log('here')
    setTimeout(()=>this.setState({addError: null}), 3000)
  }

  handleSubmit = e =>{
    e.preventDefault();
    const checkedTime =checkTime(this.state.video_length)
    if (checkedTime.error){
      this.errorHandler(checkedTime)
      return;
    }
    const video = {
      title: this.state.title,
      video_length: checkedTime.googleTimeString,
      youtube_display_name: this.state.youtube_display_name,
      tags: tagStringToArray(this.state.tags),
    }
    const isError = errorCheckNewVideo(video);
    if (isError.status === true){
      this.errorHandler(isError)
      return
    }
    VideoStorage.saveVideo(video)
    this.props.history.push('/creator')
  }
  
  handleFields = e => {
    let { value, name } = e.target;
      this.setState({
      [name]: value
    })
  }

  render() {
    return (
      <section className="add-video-container page">
        <h2 className= "provide-info-text">Please provide information on your video below</h2>
        <AddVideoForm 
          cssRouteProps = {this.cssRouteClasses}
          fields={this.state} 
          handleFields={this.handleFields}
          handleSubmit={this.handleSubmit}/>
      </section>
    );
  }
}
