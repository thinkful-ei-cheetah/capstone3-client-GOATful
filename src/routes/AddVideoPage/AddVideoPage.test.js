import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AddVideoPage from './AddVideoPage';

describe('AddVideoPage Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< AddVideoPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<AddVideoPage/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})