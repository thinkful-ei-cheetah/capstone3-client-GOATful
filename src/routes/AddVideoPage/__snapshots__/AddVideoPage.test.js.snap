// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddVideoPage Component renders the UI as expected 1`] = `
<section
  className="add-video-container page"
>
  <h2
    className="provide-info-text"
  >
    Please provide information on your video below
  </h2>
  <form
    className="add-video-form"
    onSubmit={[Function]}
  >
    <div
      className="flex-form"
    >
      <div
        className="inputs"
      >
        <label
          className="add-label"
          htmlFor="add-video-title"
        >
          Title:
        </label>
        <input
          className="add-video-input"
          id="add-video-title"
          name="title"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="fitness 100"
          required={true}
          type="text"
          value=""
        />
        <label
          className="add-label"
          htmlFor="add-video-tags"
        >
          Tags:
        </label>
        <input
          className="add-video-input"
          id="add-video-tags"
          name="tags"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="workout, fitness, gym"
          required={true}
          type="text"
          value=""
        />
        <label
          className="add-label"
          htmlFor="add-video-length"
        >
          Video Length:
        </label>
        <input
          className="add-video-input"
          id="add-video-length"
          name="video_length"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="04:13:22"
          required={true}
          type="text"
          value=""
        />
        <label
          className="add-label"
          htmlFor="add-video-youtube-name"
        >
          YouTube Name:
        </label>
        <input
          className="add-video-input"
          id="add-video-youtube-name"
          minLength="3"
          name="youtube_display_name"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="haloking"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="directions"
      >
        <h4>
          Instructions
        </h4>
        <div
          className="instructions-text"
        >
          <p
            className="leggo-text"
          >
            Let's Go!
          </p>
        </div>
      </div>
    </div>
    <button>
      Create!
    </button>
  </form>
</section>
`;
